hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -970,6 +970,27 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 	return ctrl.Result{}, nil
 }
 
+func isArchAndPlatformSupported(nodePool *hyperv1.NodePool) bool {
+	supported := false
+
+	switch nodePool.Spec.Platform.Type {
+	case hyperv1.AWSPlatform:
+		if nodePool.Spec.Arch == "amd64" || nodePool.Spec.Arch == "arm64" {
+			supported = true
+		}
+	case hyperv1.AgentPlatform:
+		if nodePool.Spec.Arch == "amd64" || nodePool.Spec.Arch == "ppc64le" {
+			supported = true
+		}
+	case hyperv1.PowerVSPlatform:
+		if nodePool.Spec.Arch == "ppc64le" {
+			supported = true
+		}
+	}
```

Comment:
We need a case for Azure - should just be `amd64`.

We need a case for Kubevirt - should just be `amd64`; correct me if I'm wrong @davidvossel.